sfe/sfe.go

```diff
@@ -0,0 +1,287 @@
+		return
+	}
+
+	err := tmplPages.ExecuteTemplate(w, subTmpl, dynamicData)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+// Index is the homepage of the SFE
+func (sfe *SelfServiceFrontEndImpl) Index(response http.ResponseWriter, request *http.Request) {
+	if request.Method != http.MethodGet && request.Method != http.MethodHead {
+		response.Header().Set("Access-Control-Allow-Methods", "GET, HEAD")
+		response.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	renderTemplate(response, "index.html", nil)
+}
+
+// BuildID tells the requester what boulder build version is running.
+func (sfe *SelfServiceFrontEndImpl) BuildID(response http.ResponseWriter, request *http.Request) {
+	response.Header().Set("Content-Type", "text/plain")
+	response.WriteHeader(http.StatusOK)
+	detailsString := fmt.Sprintf("Boulder=(%s %s)", core.GetBuildID(), core.GetBuildTime())
+	if _, err := fmt.Fprintln(response, detailsString); err != nil {
+		sfe.log.Warningf("Could not write response: %s", err)
+	}
+}
+
+// unpauseJWT is generated by a WFE and is used to round-trip back to the WFE to
+// unpause the requester's account.
+type unpauseJWT string
+
+func (sfe *SelfServiceFrontEndImpl) getHelper(response http.ResponseWriter, incomingJWT unpauseJWT) {
+	if incomingJWT != "" {
+		type tmplData struct {
+			UnpausePath string
+			JWT         string
+		}
+
+		// Serve the actual unpause page given to a Subscriber. Populates the
+		// unpause form with the JWT from the URL. That JWT itself may be
+		// invalid or expired, but that validation will be performed only after
+		// submitting the form.
+		renderTemplate(response, "unpause-params.html", tmplData{unpausePath, string(incomingJWT)})
+	} else {
+		// We only want to accept requests containing the JWT param.
+		renderTemplate(response, "unpause-noParams.html", nil)
+	}
+}
+
+// posthelper After clicking unpause, serve a page indicating if the unpause succeeded or failed.
+func (sfe *SelfServiceFrontEndImpl) postHelper(response http.ResponseWriter, incomingJWT unpauseJWT) {
+	if incomingJWT != "" {
+		type tmplData struct {
+			ShouldUnpause bool
+			AccountID     string
+		}
+
+		claims, err := sfe.validateJWTforAccount(incomingJWT)
+		if err != nil {
+			renderTemplate(response, "unpause-post.html", nil)
+		}
+
+		// TODO(#7356) Declare a registration ID variable to populate an
+		// rapb unpause account request message.
+		_, innerErr := strconv.ParseInt(claims.Subject, 10, 64)
+		if innerErr != nil {
+			renderTemplate(response, "unpause-noParams.html", nil)
+		}
+
+		// TODO(#7536) Send gRPC request to the RA informing it to unpause
+		// the account specified in the claim. At this point we should wait
+		// for the RA to process the request before returning to the client,
+		// just in case the request fails.
+
+		// Success, the account has been unpaused.
+		renderTemplate(response, "unpause-post.html", tmplData{true, claims.Subject})
```

Comment:
I'd like to see a randerTemplate() helper for each of the HTML templates which takes in exactly the parameters needed to render the template. That would make it a whole lot harder for us to accidentally miss a template variable.